### DISCUSSION TITLE
230713 Git Branch 전략 확정 회의 (16:30 ~ 17:00)

### DISCUSSION THREAD

**Discussion by somsom13** [woowacourse-teams/2023-shook#43] [회의]
## 회의 안건
Git Branch 전략을 확정한다.
- 변형된 Git Flow
- Github Flow
  - main, feat 브랜치로만 작업한다.
  - main에 반영하기 전에 feat 브랜치에서 QA를 진행한다.
  - QA 진행 시, 최신의 배포 상태를 반영하기 위해 main을 feat에 rebase 한다. (feat/2 브랜치에서 git rebase main)
- Prolog Flow

**Comment by Cyma-s** [comment]
## 브랜치 전략
- Github Flow 를 사용한다.
    - main: 배포용 브랜치. 삭제하지 않는다.
    - feat: 기능 개발용 브랜치. 기능 개발 후 삭제한다.
    - 모든 작업 브랜치는 main에서 파생된다.
    - main에 반영하기 전에 feat 브랜치에서 QA를 진행한다.
    - 개발 서버에서 QA 진행 시, 최신 배포 상태를 반영하기 위해 main을 feat에 rebase 한다. (feat/2 브랜치에서 git rebase main)
    - 현재 feat 브랜치에서 다른 feat 브랜치의 기능이 필요한 경우, feat 브랜치를 필요한 feat 브랜치에 머지한다.

```mermaid
---
title : S-HOOK Git diagram
---
%%{init: { 'theme': 'base', 'gitGraph': {'showBranches': true, 'showCommitLabel': false } } }%%
gitGraph
commit
branch feature
checkout feature
commit
commit
commit
checkout main
merge feature
branch hotfix
commit
commit
checkout main
merge hotfix
```
